Firestore/Port/ordered_code.cc

```diff
@@ -18,11 +18,17 @@
 
 #include <assert.h>
 
+#include <absl/base/internal/endian.h>
```

Comment:
I assume this file itself has been ported from absl. (Otherwise, all the 'internal' references are a bit scary.)

Reply:
This file originated in google3's //strings. It's been open sourced several times: most recently (before firestore) in [tensorflow](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/lib/strings/ordered_code.cc).

Unfortunately, this file is not (perhaps yet) in abseil. On plus side because of their common usage within google3 these macros aren't going away anytime soon. Their status in abseil is essentially that they're required but not a committed interface.

In any case, we're depending on a checked-in stable snapshot of abseil for this so we can control when we update and how to deal with it at the time. Worst case, we check-in a separate snapshot of unaligned_access.h if upstream completely breaks this.